FindNearby/force-app/main/default/lwc/mobileMap/mobileMap.js

```diff
@@ -0,0 +1,152 @@
+    listView = '';
+    mapOptions = {
+        disableDefaultUI: true,
+        zoomControl: false
+    };
+
+    renderedCallback() {
+        if (!this.isMarkersListEqual()) {
+            this.markersWithResource =
+                this.resourceMarker ?
+                [...this.filteredMarkers, this.resourceMarker] :
+                [...this.filteredMarkers];
+        }
+    }
+
+    isMarkersListEqual() {
+        const markers = this.markersWithResource.filter(m => m.value !== undefined);
+        return (
+            this.filteredMarkers.length === markers.length &&
+            this.filteredMarkers.every((m, ind) => m.value.id === markers[ind].value.id)
+        );
+    }
+
+    @api setResourceMarker({ lat, lng }) {
+        this.resourceMarker = {
+            location: { Latitude: lat.substring(0, 10), Longitude: lng.substring(0, 10) },
+            mapIcon: this.getResourceMarkerSVG()
+        };
+    }
+
+    onMarkerSelect(event) {
+        try {
+            if (!event.target.selectedMarkerValue) return;
+            const ind = this.filteredMarkers.findIndex(m => m.value.id === event.target.selectedMarkerValue.id);
+            this.setCurrentMarker(ind);
+        } catch (error) {
+            this.handleError(error);
+        }
+    }
+
+    // Buttons Handlers
+
+    handleLocatorClick() {
+        this.located = true;
+        this.setCardState(false);
+    }
+
+    handleCloseCardClick() {
+        this.setCardState(false);
+    }
+
+    handleRedirectClick = () => {
+        this.redirectToMarkerDetails();
+    }
+
+    handleRouteClick = () => {
+        this.routeToMarkerLocation();
+    }
+
+    handleBackCardClick() {  
+        let ind = this.currentMarkerInd - 1;
+        if (ind < 0) ind = this.filteredMarkers.length - 1;
+        this.setCurrentMarker(ind);
+    }
+
+    handleNextCardClick() {
+        const ind = (this.currentMarkerInd + 1) % this.filteredMarkers.length;
+        this.setCurrentMarker(ind);
+    }
+
+    // Getters & Helpers
+
+    get center() {
+        return this.located ? this.resourceMarker : this.currentMarker;
+    }
+
+    get currentMarkerTitle() {
+        return this.currentMarker?.value?.title;
+    }
+
+    get currentMarkerDetailFieldName() {
+        return this.currentMarker?.value?.detailFieldName;
+    }
+
+    get currentMarkerDetailFieldValue() {
+        return this.currentMarker?.value?.detailFieldValue;
+    }
+
+    get currentMarkerIndexUi() {
+        const indUi = this.currentMarkerInd + 1;
+        return isNaN(indUi) ? 0 : indUi;
+    }
+
+    get currentMarkerDistance() {
+        return this.currentMarker?.value?.distance;
+    }
+
+    get currentMarkerIconUrl() {
+        return this.currentMarker?.value?.iconUrl;
+    }
+
+    setCardState = (state) => {
+        this.showCard = state;
+    }
+
+    getResourceMarkerSVG() {
+        const icon =
+            `<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
```

Comment:
just curious: would it not be possible to have this icon as a static resource of `FindNearby` project rather than a hardcoded string here?

Reply:
I've tried this for a long time, I still have my last attempt in stash, but couldn't make it work.